bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -164,8 +164,15 @@ private MergeSource(String repositoryName, String branchName) {
             if (commits.size() < 2) {
                 ret.add("A Merge PR must contain at least two commits that are not already present in the target.");
             } else {
-                if (!commits.get(0).isMerge()) {
-                    ret.add("The top commit must be a merge commit.");
+                // Find the last merge commit - the very last commit is not eligible (as the merge needs a parent)
+                int mergeCommitIndex = commits.size();
+                for (int i = 0; i < commits.size() - 1; ++i) {
+                    if (commits.get(i).isMerge()) {
+                        mergeCommitIndex = i;
+                    }
+                }
+                if (mergeCommitIndex >= commits.size() - 1) {
```

Comment:
What if you do a "Merge PR" with a single trivial merge just to merge to branches (that have identical commits). Too narrow of a use case to think about?

Reply:
Not sure if that is something we would need to support, not even sure how to create such a merge commit. :)

Reply:
Ok, then forget about this comment :smile: